lib/decodeOpaqueId.js

```diff
@@ -8,16 +8,15 @@
  */
 export default function decodeOpaqueId(opaqueId) {
   if (opaqueId === undefined || opaqueId === null) return null;
-  const buffer = Buffer.from(opaqueId, "base64");
 
-  // As far as I can tell, round-tripping back to base64 is the
-  // only robust way to determine whether it was encoded as base64
-  // in the first place.
-  if (buffer.toString("base64") !== opaqueId) {
-    return { namespace: null, id: opaqueId };
+  const [namespace, id] = Buffer
+    .from(opaqueId, "base64")
+    .toString("utf8")
+    .split(":");
```

Comment:
We should pass  2 as the `limit` to `.split(":",  2)` for performance and correctness if someone encodes an id with more than 1 colon char in it.